client/main.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { Provider } from 'react-redux'
+import { Router, createMemoryHistory, browserHistory } from 'react-router'
+import createStore from 'store/createStore'
+import Styletron from 'styletron-client'
+import { StyletronProvider } from 'styletron-react'
+// import AppContainer from './components/App'
+import injectTapEventPlugin from 'react-tap-event-plugin'
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
+import { polyfill as promisePolyfill } from 'es6-promise'
+import 'isomorphic-fetch'
+
+const styleElements = document.getElementsByClassName('_styletron_hydrate_')
+
+promisePolyfill()
+
+// ========================================================
+// Store Instantiation
+// ========================================================
+
+const initialState = window.__PRELOADED_STATE__
+const store = createStore(initialState)
```

Comment:
I think we should use immutable structure by default. Helps a lot of react optimization and clean reducer code